src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AmbientValueAttribute.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+
+
+
+/*
+ */
+
+namespace System.ComponentModel
+{
+    using System;
+    using System.ComponentModel;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+    //using System.Runtime.Serialization.Formatters;
+    using System.Security.Permissions;
+
+    /// <summary>
+    ///    <para>Specifies the ambient value for a property.  The ambient value is the value you
+    ///    can set into a property to make it inherit its ambient.</para>
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1019:DefineAccessorsForAttributeArguments")]
+    [AttributeUsage(AttributeTargets.All)]
+    public sealed class AmbientValueAttribute : Attribute
+    {
+        private readonly object _value;
+
+        /// <summary>
+        /// <para>Initializes a new instance of the <see cref='System.ComponentModel.AmbientValueAttribute'/> class, converting the
+        ///    specified value to the
+        ///    specified type, and using the U.S. English culture as the
+        ///    translation
+        ///    context.</para>
+        /// </summary>
+        public AmbientValueAttribute(Type type, string value)
+        {
+            // The try/catch here is because attributes should never throw exceptions.  We would fail to
+            // load an otherwise normal class.
+            try
+            {
+                _value = TypeDescriptor.GetConverter(type).ConvertFromInvariantString(value);
+            }
+            catch
+            {
+                Debug.Fail("Ambient value attribute of type " + type.FullName + " threw converting from the string '" + value + "'.");
```

Comment:
Nit: for readability in such asserts, I personally prefer string interpolation, e.g.
```C#
Debug.Fail($"Ambient value attribute of type {type.FullName} threw converting from the string '{value}'");
```